packages/fxa-settings/src/components/Settings/ConnectedServices/index.tsx

```diff
@@ -106,7 +106,8 @@ export const ConnectedServices = forwardRef<HTMLDivElement>((_, ref) => {
         // disconnect all clients/sessions with this name since only unique names
         // are displayed to the user. This is batched into one network request
         // via BatchHttpLink
-        const clientsWithMatchingName = groupedByName[client.name || 'unknown'];
+        const groupByKey = client.name || '';
+        const clientsWithMatchingName = groupedByName[groupByKey];
```

Comment:
What happens if there are multiple clients where the name is undefined? I'm guess, this won't work here? Seems like we need loop over the attachedClient list directly?

Reply:
In this case, all the unknown clients would be disconnected. In my testing only clients that failed to parse a user agent had an undefined name. OAuth clients set this value to the name of the RP and sync uses the device record. 

Maybe this is correct to disconnect all unknown clients?